airflow/providers_manager.py

```diff
@@ -85,10 +84,9 @@ def ensure_prefix(field):
 
 
 if TYPE_CHECKING:
-    from typing_extensions import Literal
-
     from airflow.decorators.base import TaskDecorator
     from airflow.hooks.base import BaseHook
+    from airflow.typing_compat import Literal
```

Comment:
Why do you import it from airflow.typing_compact and not from typing_extension?

Reply:
Short version: We decided to use `airflow.typing_compat` into the core. Providers still directly use `typing_extension`

https://github.com/apache/airflow/pull/33821